components/plan-build/bin/hab-plan-build.sh

```diff
@@ -2521,117 +2512,101 @@ pkg_artifact="$HAB_CACHE_ARTIFACT_PATH/${pkg_origin}-${pkg_name}-${pkg_version}-
 _find_system_commands
 _determine_hab_bin
 
-case "${pkg_type}" in
-    "composite")
-        source "${source_dir}/composite_build_functions.bash"
-
-        mkdir -pv "$CACHE_PATH"
-        mkdir -pv "$pkg_prefix"
-
-        # Preliminaries
-        _setup_composite_build_global_variables
-
-        _validate_composite
-        _render_composite_metadata
-        ;;
-    "standalone")
-        # Non-composite ("normal") Habitat package
-
-        # Run `do_begin`
-        build_line "$_program setup"
-        do_begin
+# We removed support for composite plans in habitat 0.76.  Let's give a useful error message
+# if you happen to try and build one rather than create a useless package
+if [[ -n "${pkg_type:-}" && "${pkg_type}" == "composite" ]]; then
```

Comment:
To confirm: this would only happen if a plan had a `pkg_type` defined, since we've otherwise removed that variable from this file, correct?

Reply:
Yep.  Here are the three cases:

* With `pkg_type="composite" 
```sh
$ grep pkg_type foo/habitat-composite-example/composite-example/plan.sh
pkg_type="composite"
$  `uname' detection
   : Loading /home/james/src/github/jamesc/habitat/foo/habitat-composite-example/composite-example/plan.sh
   composite-example: Plan loaded
   composite-example: Validating plan metadata
   composite-example: Using HAB_BIN=/bin/hab for installs, signing, and hashing
   composite-example: ERROR: Composite plans are no longer supported. For more details see https://www.habitat.sh/blog/2018/10/shelving-composites/
   composite-example: Build time: 0m0s
   composite-example: Exiting on error
```
* with `pkg_type="standalone"
```sh
$ grep pkg_type foo/habitat-composite-example/sample-node-app/habitat/plan.sh
pkg_type="standalone"
$ ./components/plan-build/bin/hab-plan-build.sh  foo/habitat-composite-example/sample-node-app/
   : Setting pkg_target='x86_64-linux' using `uname' detection
   : Loading /home/james/src/github/jamesc/habitat/foo/habitat-composite-example/sample-node-app/habitat/plan.sh
   sample-node-app: Plan loaded
   sample-node-app: Validating plan metadata
   sample-node-app: Using HAB_BIN=/bin/hab for installs, signing, and hashing
   sample-node-app: hab-plan-build.sh setup
   sample-node-app: Writing pre_build file
   ...
   ...
```

* With nothing set

```sh
$ grep pkg_type foo/habitat-composite-example/sample-node-app/habitat/plan.sh
$  ./components/plan-build/bin/hab-plan-build.sh  foo/habitat-composite-example/sample-node-app
   : Setting pkg_target='x86_64-linux' using `uname' detection
   : Loading /home/james/src/github/jamesc/habitat/foo/habitat-composite-example/sample-node-app/habitat/plan.sh
   sample-node-app: Plan loaded
   sample-node-app: Validating plan metadata
   sample-node-app: Using HAB_BIN=/bin/hab for installs, signing, and hashing
   sample-node-app: hab-plan-build.sh setup
   sample-node-app: Writing pre_build file
   ...
   ...
```